core-wih/src/main/java/io/elimu/a2d2/corewih/OAuth2WorkItemHandler.java

```diff
@@ -0,0 +1,122 @@
+package io.elimu.a2d2.corewih;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.io.PrintStream;
+import java.io.StringWriter;
+import java.net.URL;
+import java.util.ArrayList;
+import java.util.Base64;
+import java.util.List;
+import java.util.Map;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+import javax.net.ssl.HttpsURLConnection;
+
+import org.kie.api.runtime.process.WorkItem;
+import org.kie.api.runtime.process.WorkItemHandler;
+import org.kie.api.runtime.process.WorkItemManager;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import io.elimu.a2d2.exception.WorkItemHandlerException;
+
+public class OAuth2WorkItemHandler implements WorkItemHandler {
+
+	private static final Logger LOG = LoggerFactory.getLogger(OAuth2WorkItemHandler.class);
+	
+	private final Pattern patToken = Pattern.compile(".*\"access_token\"\\s*:\\s*\"([^\"]+)\".*");
+	private final Pattern patExpires = Pattern.compile(".*\"expires_in\"\\s*:\\s*([^,]+).*");
+
+	@Override
+	public void executeWorkItem(WorkItem workItem, WorkItemManager manager) {
+		String clientId = (String) workItem.getParameter("clientId");
+		String scope = (String) workItem.getParameter("scope");
+		String username = (String) workItem.getParameter("username");
+		String password = (String) workItem.getParameter("password");
+		String grantType = (String) workItem.getParameter("grantType");
+		String tokenUrl = (String) workItem.getParameter("tokenUrl");
+		List<String> missingValues = new ArrayList<>();
+		if (clientId == null || "".equals(clientId.trim())) {
```

Comment:
Consider using `StringUtils.isEmpty(str)` from Apache Commons to clean up these checks.

Reply:
Changed to StringUtils.isEmpty